homeassistant/components/ipma/weather.py

```diff
@@ -68,71 +67,82 @@
 
+async def async_get_api(hass):
+    """Get the pyipma api object."""
     websession = async_get_clientsession(hass)
+    return IPMA_API(websession)
+
+
+async def async_get_location(hass, api, latitude, longitude):
+    """Retrieve pyipma location, location name to be used as the entity name."""
     with async_timeout.timeout(10):
-        station = await Station.get(websession, float(latitude), float(longitude))
+        location = await Location.get(api, float(latitude), float(longitude))
 
     _LOGGER.debug(
         "Initializing for coordinates %s, %s -> station %s",
         latitude,
         longitude,
-        station.local,
+        location.station,
     )
 
-    return station
+    return location
 
 
 class IPMAWeather(WeatherEntity):
     """Representation of a weather condition."""
 
-    def __init__(self, station, config):
+    def __init__(self, location: Location, api: IPMA_API, config):
         """Initialise the platform with a data instance and station name."""
-        self._station_name = config.get(CONF_NAME, station.local)
-        self._station = station
-        self._condition = None
+        self._api = api
+        self._location_name = config.get(CONF_NAME, location.name)
+        self._location = location
+        self._observation = None
         self._forecast = None
-        self._description = None
 
     @Throttle(MIN_TIME_BETWEEN_UPDATES)
     async def async_update(self):
         """Update Condition and Forecast."""
         with async_timeout.timeout(10):
-            _new_condition = await self._station.observation()
-            if _new_condition is None:
-                _LOGGER.warning("Could not update weather conditions")
-                return
-            self._condition = _new_condition
+            new_observation = await self._location.observation(self._api)
+            new_forecast = await self._location.forecast(self._api)
+
+            if new_observation:
+                self._observation = new_observation
+            else:
+                _LOGGER.warning("Could not update weather observation")
+
+            if new_forecast:
+                self._forecast = [f for f in new_forecast if f.forecasted_hours == 24]
+            else:
+                _LOGGER.warning("Could not update weather forecast")
 
             _LOGGER.debug(
-                "Updating station %s, condition %s",
-                self._station.local,
-                self._condition,
+                "Updated location %s, observation %s",
+                self._location.name,
+                self._observation,
             )
-            self._forecast = await self._station.forecast()
-            self._description = self._forecast[0].description
 
     @property
     def unique_id(self) -> str:
         """Return a unique id."""
-        return f"{self._station.latitude}, {self._station.longitude}"
+        return f"{self._location.global_id_local}, {self._location.id_station}"
```

Comment:
Changing the unique_id is a breaking change. Please note it in the PR description. Users may need to clear out stale entities from the entity registry.

Reply:
I think @abmantis can avoid the breaking change and keep the old ID

Reply:
Yes, I think I can keep the old id's.